### PR TITLE
 Navigation: Fade in navigation for DynamicSection when scrolling down

### PR DESCRIPTION
**Pull request by kamransaleem**
#2637

### CODE REVIEW

**Comment by davidwatkins73** [comment]
@rovats - could you also review - think we both need to be familiar with this code

File: waltz-ng/client/common/components/dynamic-section-navigation/dynamic-section-navigation.scss [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+@import "../../../../style/_variables";
+@import "../../../../style/_util";
+
+
+.waltz-dynamic-section-navigation ul {
+    list-style-type:none;
+    /*height:30px;*/
```
Comment (davidwatkins73) [inline waltz-ng/client/common/components/dynamic-section-navigation/dynamic-section-navigation.scss]: why commented ?

File: waltz-ng/client/common/components/dynamic-section-navigation/dynamic-section-navigation.scss [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    left: 0px;
+    z-index: 999;
+    border-bottom: 2px solid $waltz-border-color-lighter;
+    opacity: 0;
+
+    padding-top: 2px;
+    padding-left: 36px;
+    padding-right: 36px;
+
+    @include transition(opacity 0.2s);
```
Comment (davidwatkins73) [inline waltz-ng/client/common/components/dynamic-section-navigation/dynamic-section-navigation.scss]: may be worth using `$animation-speed` from `_variables.scss`

File: waltz-ng/client/common/components/dynamic-section-navigation/dynamic-section-navigation.scss [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+.waltz-dynamic-section-navigation.wdsn-sticky.wdsn-sticky-visible {
+    opacity: 1;
+    @include transition(opacity 0.2s);
```
Comment (davidwatkins73) [inline waltz-ng/client/common/components/dynamic-section-navigation/dynamic-section-navigation.scss]: may be worth using `$animation-speed` from `_variables.scss`

File: waltz-ng/client/common/index.js [hunk 4]
```diff
@@ -174,3 +174,14 @@ export function invokeFunction(fn) {
...
+export function scrollTo($interval, $window, yOffset, amount = 50, direction = -1) {
+    const jump = amount * direction;
+    const timerPromise = $interval(() => {
+        $window.scrollBy(0, jump);
+        if ($window.pageYOffset <= yOffset)
+            $interval.cancel(timerPromise);
+    }, 10);
+    return timerPromise;
+}
+
```
Comment (davidwatkins73) [inline waltz-ng/client/common/index.js]: is this code mimicing smooth scrolling ? May be worth a bit of jsdoc.

File: waltz-ng/client/common/index.js [hunk 5]
```diff
@@ -174,3 +174,14 @@ export function invokeFunction(fn) {
 }
 
 
+export function scrollTo($interval, $window, yOffset, amount = 50, direction = -1) {
+    const jump = amount * direction;
+    const timerPromise = $interval(() => {
+        $window.scrollBy(0, jump);
+        if ($window.pageYOffset <= yOffset)
```
Comment (rovats) [inline waltz-ng/client/common/index.js]: This probably won't work if direction is 1 (scrolling down)..

**Pull request merged**